cmd/gctcli/orderbook.go

```diff
@@ -0,0 +1,744 @@
+		return errInvalidAsset
+	}
+
+	p, err := currency.NewPairDelimiter(currencyPair, pairDelimiter)
+	if err != nil {
+		return err
+	}
+
+	var percentage float64
+	if c.IsSet("asset") {
+		percentage = c.Float64("percent")
+	} else {
+		percentage, _ = strconv.ParseFloat(c.Args().Get(3), 64)
+	}
+
+	conn, cancel, err := setupClient(c)
+	if err != nil {
+		return err
+	}
+	defer closeConn(conn, cancel)
+
+	client := gctrpc.NewGoCryptoTraderServiceClient(conn)
+	result, err := client.GetOrderbookAmountByNominal(c.Context,
+		&gctrpc.GetOrderbookAmountByNominalRequest{
+			Exchange: exchangeName,
+			Pair: &gctrpc.CurrencyPair{
+				Base:  p.Base.String(),
+				Quote: p.Quote.String(),
+			},
+			Asset:             assetType,
+			Sell:              isSelling,
+			NominalPercentage: percentage,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	jsonOutput(result)
+	return nil
+}
+
+var impact = &cli.Command{
+	Name:      "impact",
+	Usage:     "simulates a buy or sell based off the reference price and the orderbook impact slippage.",
+	ArgsUsage: "<exchange> <pair> <asset> <percent>",
+	Action:    getImpact,
+	Flags: append(orderbookCommonFlags, &cli.Float64Flag{
+		Name:  "percent",
+		Usage: "the max percentage slip you wish to occur e.g. 1 = 1% and 100 = 100%. Note: If selling base/hitting the bids you can only have a max value of 100%",
+	}),
+}
+
+func getImpact(c *cli.Context) error {
+	isSelling := c.Bool("sell")
+	if c.NArg() == 0 && c.NumFlags() == 0 {
+		return cli.ShowCommandHelp(c, "impact")
+	}
+
+	var exchangeName string
+	if c.IsSet("exchange") {
+		exchangeName = c.String("exchange")
+	} else {
+		exchangeName = c.Args().First()
+	}
+
+	var currencyPair string
+	if c.IsSet("pair") {
+		currencyPair = c.String("pair")
+	} else {
+		currencyPair = c.Args().Get(1)
+	}
+
+	if !validPair(currencyPair) {
+		return errInvalidPair
+	}
+
+	var assetType string
+	if c.IsSet("asset") {
+		assetType = c.String("asset")
+	} else {
+		assetType = c.Args().Get(2)
+	}
+
+	assetType = strings.ToLower(assetType)
+	if !validAsset(assetType) {
+		return errInvalidAsset
+	}
+
+	p, err := currency.NewPairDelimiter(currencyPair, pairDelimiter)
+	if err != nil {
+		return err
+	}
+
+	var percentage float64
+	if c.IsSet("asset") {
+		percentage = c.Float64("percent")
+	} else {
+		percentage, _ = strconv.ParseFloat(c.Args().Get(3), 64)
+	}
+
+	conn, cancel, err := setupClient(c)
+	if err != nil {
+		return err
+	}
+	defer closeConn(conn, cancel)
+
+	client := gctrpc.NewGoCryptoTraderServiceClient(conn)
+	result, err := client.GetOrderbookAmountByImpact(c.Context,
+		&gctrpc.GetOrderbookAmountByImpactRequest{
+			Exchange: exchangeName,
+			Pair: &gctrpc.CurrencyPair{
+				Base:  p.Base.String(),
+				Quote: p.Quote.String(),
+			},
+			Asset:            assetType,
+			Sell:             isSelling,
+			ImpactPercentage: percentage,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	jsonOutput(result)
+	return nil
+}
+
+var quote = &cli.Command{
+	Name:      "quote",
```

Comment:
Another comment on this one. Is it possible to infer pricing? I'll run you through what I'm getting at.
I performed the following command:
```
go run . orderbook whalebomb --exchange="ftx" --asset="futures" --pair="btc-perp" --price="21790" --side="buy"

 "orders": [
   ...
 ],
 "amount": 3737841.4552,
 "minimum_price": 21774,
 "maximum_price": 21790,
 "percentage_gain_loss": 0.07348213465601176,
 "status": "Buying 3737841.46 PERP worth of BTC will send the price from 21774 to 21790 [0.07%] and take 16 orders. "
}
```
From here, it tells me, in order to move the price, I need to spend `3737841.4552`
So then I do the following:
```
go run . orderbook buy quote ftx btc-perp futures 3737841.4552
{
 "nominal_percentage": 0.03983502546274574,
 "impact_percentage": 0.07348213465601176,
 "slippage_cost": 1488.3771999995224,
 "currency_bought": "BTC",
 "currency_sold": "PERP",
 "side_affected": "ASK",
 "update_protocol": "REST"
}
```

What I see here, is that the `impact_percentage` matches the `percent_gain_loss`. Hurray! :tada: , but what is missing, is the price that it roughly lands at, which is about 21790 `PERP`. Is the pricing something that is known and can be added? Or, am I misusing this command, and as a result, should just stick to whalebomb? Because as the description says, this command is simulating a buy using quote, so it could theoretically, say where the price has gone as a result of this buy (which I believe is the impact percentage)